### PR TITLE
send customizationArn in metric clientComponentLatency, perceivedLatency

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
https://github.com/aws/aws-toolkit-common/pull/738

Will revert telemetry override when ^ is merged

image

## builderId

```
DefaultMetricEvent(createTime=2024-05-22T18:11:48.898972Z, awsAccount=168976375234, awsRegion=us-east-1, awsProduct=Amazon Q For JetBrains, awsVersion=3.5-SNAPSHOT-233+unknownCommit, data=[DefaultDatum(name=codewhisperer_perceivedLatency, value=1.0, unit=None, passive=true, metadata={credentialStartUrl=https://view.awsapps.com/start, duration=890.0, codewhispererLanguage=java, codewhispererUserGroup=Control, codewhispererCompletionType=Line, codewhispererRequestId=dcfbe348-2a12-4976-b8be-cc2ece0b2417, codewhispererSessionId=8370c35e-2cb7-4f75-94f2-bf6043e63058, codewhispererTriggerType=OnDemand})])
DefaultMetricEvent(createTime=2024-05-22T18:11:49.066137Z, awsAccount=168976375234, awsRegion=us-east-1, awsProduct=Amazon Q For JetBrains, awsVersion=3.5-SNAPSHOT-233+unknownCommit, data=[DefaultDatum(name=codewhisperer_clientComponentLatency, value=1.0, unit=None, passive=true, metadata={codewhispererCredentialFetchingLatency=0.0, codewhispererUserGroup=Control, codewhispererPostprocessingLatency=17.0, codewhispererRequestId=dcfbe348-2a12-4976-b8be-cc2ece0b2417, codewhispererAllCompletionsLatency=726.0, codewhispererPreprocessingLatency=1.0, codewhispererSessionId=8370c35e-2cb7-4f75-94f2-bf6043e63058, credentialStartUrl=https://view.awsapps.com/start, codewhispererLanguage=java, codewhispererFirstCompletionLatency=545.0, codewhispererEndToEndLatency=565.0, codewhispererCompletionType=Line, codewhispererTriggerType=OnDemand})])

```

## IdC with customization
```
DefaultMetricEvent(createTime=2024-05-22T18:09:17.336613Z, awsAccount=168976375234, awsRegion=us-east-1, awsProduct=Amazon Q For JetBrains, awsVersion=3.5-SNAPSHOT-233+unknownCommit, data=[DefaultDatum(name=codewhisperer_perceivedLatency, value=1.0, unit=None, passive=true, metadata={credentialStartUrl=https://amzn.awsapps.com/start, duration=398538.0, codewhispererLanguage=java, codewhispererUserGroup=Control, codewhispererCompletionType=Line, codewhispererRequestId=49ed68fc-75f3-4eeb-8e81-31e7fdced1d1, codewhispererCustomizationArn=arn:aws:codewhisperer:us-east-1:724904587780:customization/HVM3VQGKG7QP, codewhispererSessionId=1f8586fb-fa93-4fc4-b49c-aa8ef1a964f2, codewhispererTriggerType=OnDemand})])
DefaultMetricEvent(createTime=2024-05-22T18:09:17.773066Z, awsAccount=168976375234, awsRegion=us-east-1, awsProduct=Amazon Q For JetBrains, awsVersion=3.5-SNAPSHOT-233+unknownCommit, data=[DefaultDatum(name=codewhisperer_clientComponentLatency, value=1.0, unit=None, passive=true, metadata={codewhispererCredentialFetchingLatency=1.0, codewhispererUserGroup=Control, codewhispererPostprocessingLatency=67.0, codewhispererRequestId=49ed68fc-75f3-4eeb-8e81-31e7fdced1d1, codewhispererAllCompletionsLatency=1758.0, codewhispererPreprocessingLatency=122.0, codewhispererCustomizationArn=arn:aws:codewhisperer:us-east-1:724904587780:customization/HVM3VQGKG7QP, codewhispererSessionId=1f8586fb-fa93-4fc4-b49c-aa8ef1a964f2, credentialStartUrl=https://amzn.awsapps.com/start, codewhispererLanguage=java, codewhispererFirstCompletionLatency=1262.0, codewhispererEndToEndLatency=1452.0, codewhispererCompletionType=Line, codewhispererTriggerType=OnDemand})])

```

## IdC without customization

```
DefaultMetricEvent(createTime=2024-05-22T18:10:59.444380Z, awsAccount=168976375234, awsRegion=us-east-1, awsProduct=Amazon Q For JetBrains, awsVersion=3.5-SNAPSHOT-233+unknownCommit, data=[DefaultDatum(name=codewhisperer_perceivedLatency, value=1.0, unit=None, passive=true, metadata={credentialStartUrl=https://amzn.awsapps.com/start, duration=876.0, codewhispererLanguage=java, codewhispererUserGroup=Control, codewhispererCompletionType=Line, codewhispererRequestId=e21bc8a5-c983-4d46-b68f-94efbb2dec3c, codewhispererSessionId=0cb47b2f-f59b-448b-85e5-f5178514e737, codewhispererTriggerType=OnDemand})])
DefaultMetricEvent(createTime=2024-05-22T18:10:59.445020Z, awsAccount=168976375234, awsRegion=us-east-1, awsProduct=Amazon Q For JetBrains, awsVersion=3.5-SNAPSHOT-233+unknownCommit, data=[DefaultDatum(name=codewhisperer_clientComponentLatency, value=1.0, unit=None, passive=true, metadata={codewhispererCredentialFetchingLatency=0.0, codewhispererUserGroup=Control, codewhispererPostprocessingLatency=24.0, codewhispererRequestId=e21bc8a5-c983-4d46-b68f-94efbb2dec3c, codewhispererAllCompletionsLatency=612.0, codewhispererPreprocessingLatency=3.0, codewhispererSessionId=0cb47b2f-f59b-448b-85e5-f5178514e737, credentialStartUrl=https://amzn.awsapps.com/start, codewhispererLanguage=java, codewhispererFirstCompletionLatency=612.0, codewhispererEndToEndLatency=640.0, codewhispererCompletionType=Line, codewhispererTriggerType=OnDemand})])

```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/resources/telemetryOverride.json
Comment (rli) [inline plugins/core/jetbrains-community/resources/telemetryOverride.json:714]: common?

Comment (Will-ShaoHua) [reply plugins/core/jetbrains-community/resources/telemetryOverride.json:714]: yea i'm waiting for approval https://github.com/aws/aws-toolkit-common/pull/738 lol

Comment (Will-ShaoHua) [reply plugins/core/jetbrains-community/resources/telemetryOverride.json:714]: will update when it's merged

**Comment by Will-ShaoHua** [comment]
irrelevant to this PR. 
These tests are flaky and need to be fixed

image

File: plugins/core/jetbrains-community/resources/telemetryOverride.json
Comment (Will-ShaoHua) [reply plugins/core/jetbrains-community/resources/telemetryOverride.json]: donedone

**Review by andrewyuq — APPROVED** [review]

**Review by rli — APPROVED** [review]

**Pull request merged**
